doc/api/fs.md

```diff
@@ -283,6 +283,97 @@ synchronous use libuv's threadpool, which can have surprising and negative
 performance implications for some applications. See the
 [`UV_THREADPOOL_SIZE`][] documentation for more information.
 
+## Class: fs.DirectoryEntry
+<!-- YAML
+added: REPLACEME
+-->
+
+When [`fs.readdir()`][] or [`fs.readdirSync()`][] is called with the
+`withFileTypes` option set to `true`, the resulting array is filled with
+`fs.DirectoryEntry` objects, rather than strings or `Buffers`.
+
+Note that on some systems, getting file types at the same time as `readdir` is
+not supported. On those systems, only `dirent.isUnknown()` will return true, and
```

Comment:
The mix of `DirectoryEntry` and `dirent` got me.
If we call it `Stat` and not `Status` maybe `DirectoryEntry` should be called `Dirent`.

Reply:
@jdalton I'm not sure we gain much from brevity, since users don't ever even need to use the class name in order to use this feature. What if I just change `dirent` to `directoryentry`?